### PR TITLE
Handle torch.override.is_tensor_like

### PR DESCRIPTION
**Pull request by anijain2305**
To handle new changes in pytorch.distributions

### CODE REVIEW

File: torchdynamo/variables/tensor.py
Comment (jansel) [inline torchdynamo/variables/tensor.py:114]: What is the motivation for putting this in the graph.  Could we do any useful computation in it?

This could just be a ConstantVariable with a guard?  (And not go into the graph.)

I haven't seen the use case, so just curious.

Comment (anijain2305) [reply torchdynamo/variables/tensor.py:114]: One example is Normal distribution - https://github.com/pytorch/pytorch/blob/master/torch/distributions/normal.py#L45

The inputs to this init (loc and scale) are tensors. That's the main reason, I put it in the graph. 

`is_tensor_like` is used here - https://github.com/pytorch/pytorch/blob/master/torch/distributions/utils.py#L11

But, maybe since this is called in __init__, it should be ok to be ConstantVariable?

More generally, how do you decide what to put as Constant vs DynamicShape? For example, would you put `torch.seed()` as ConstantVariable and put a guard on it?

Comment (jansel) [reply torchdynamo/variables/tensor.py:114]: In those examples the result seems to feed directly into control flow.

I'd suggest either a constant or a graph break (by adding this to disallowed functions).

The basic heuristic I use is to think about if the compiler could do anything useful with it.

`torch.seed()` should *not* be a constant, because we don't want to specialize on it.  It is possibly some random value.

`is_tensor_like(Tensor)` seems like it should just always be true.  We already guard on the fact the Tensor is a Tensor.

Comment (anijain2305) [reply torchdynamo/variables/tensor.py]: Makes sense. Making it a `ConstantVariable`.

Comment (anijain2305) [reply torchdynamo/variables/tensor.py]: IIUC, specializing on `torch.seed()` would be bad, because we would be causing so many recompilations because `torch.rand()` will most possibly be some random value.

But `is_tensor_like` is mostly going to be true, so it is ok to specialize on that. Even if it changes, we will most likely do a couple of recompilations and that will be good enough to cover all the scenarios. And we already have guard on Tensor, so we are not adding any more guards as well. 

Is that the right way to think about it?

Comment (jansel) [reply torchdynamo/variables/tensor.py]: Yes.  Also is_tensor_like() usually flows into control flow, so making it a constant allows bigger graphs.

File: torchdynamo/variables/tensor.py
Comment (jansel) [inline torchdynamo/variables/tensor.py:113]: This is the wrong place to make it a constant.  At this point we already added a node to the FX graph, so it is too late.

You could handle this similar to `torch.is_tensor` which is here:
https://github.com/facebookresearch/torchdynamo/blob/1bc7be4de849aeb3aacee1259bfb593fce79a137/torchdynamo/variables/torch.py#L104

**Review by jansel — CHANGES_REQUESTED** [review]

File: torchdynamo/variables/tensor.py
Comment (jansel) [reply torchdynamo/variables/tensor.py:113]: L104 handles the True case.  For the False case you can add it here:
https://github.com/facebookresearch/torchdynamo/blob/1bc7be4de849aeb3aacee1259bfb593fce79a137/torchdynamo/variables/torch.py#L72

**Comment by anijain2305** [comment]
Thanks for the comments. It makes sense now.

Incorporated your changes, and also added a test.

**Review by jansel — APPROVED** [review]
lgtm, feel free to merge once merge conflict is fixed

**Pull request merged**
